alacritty/src/input.rs

```diff
@@ -324,6 +324,94 @@ impl<T: EventListener> Execute<T> for Action {
     }
 }
 
+impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
+    fn xterm_modifiers(&mut self) -> u8 {
+        let mods = self.ctx.modifiers();
+        let mut n = 0;
+
+        if mods.shift() {
+            n |= 1;
+        }
+
+        if mods.alt() {
+            n |= 2;
+        }
+
+        if mods.ctrl() {
+            n |= 4;
+        }
+
+        n + 1
+    }
+
+    fn process_received_char(&mut self, c: char) {
+        let len = c.len_utf8();
+
+        if len == 1
+            && !self.ctx.modifiers().is_empty()
+            && self.ctx.terminal().mode().contains(TermMode::MODIFY_OTHER_KEYS1)
+        {
+            // xterm's definition of `other` in `modifyOtherKeys` is somewhat
+            // vague, mostly stemming out of xterm's inner design:
+            //
+            // https://github.com/ThomasDickey/xterm-snapshots/blob/f8a406455c18ec6f3de1cf2c642f7b7f0312d362/input.c#L547
+            //
+            // That's why our implementation in here is a mere approximation -
+            // it was done by analyzing "what xterm does" and trying to shoehorn
+            // it in here.
+            //
+            // To avoid duplicating comments, the implementation below follows
+            // as pure code, with comments extracted into tests at th bottom of
+            // this file.
+
+            let key = c as u8;
+            let xmods = self.xterm_modifiers();
+            let mods = self.ctx.modifiers();
+
+            let mut allowed = c == '-' || c == '=' || c == ',' || c == '.' || c.is_ascii_digit();
```

Comment:
This doesn't make much sense, does it? Most of these aren't "other" keys at all.

Reply:
Well, to be honest what I've done is to mimic xterm behavior - and after activating modifyOtherKeys=1, xterm _does_ send new escape codes for those characters; but yeah, the definition of "other" is pretty vague here.

> I'd be curious if there's any "good citizen" terminal application that actually has a useful application for this.

vim? 😅 

Out of other solutions, there's also https://sw.kovidgoyal.net/kitty/keyboard-protocol.html - overall it's a better approach, but I'm not sure about its compatibility; I'll check later - if tmux and some other applications implement it, it might be more profitable one to implement than modifyOtherKeys.

Reply:
Actually, `CSI u` seems to be supported quite nicely - I'll try implementing that one instead.

Reply:
`CSI u` is a much more reasonable approach to solving this and other problems.